kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/validation/JsonSchemaRecordValidationIT.java

```diff
@@ -249,7 +249,48 @@ void clientSideUsesSchemasToo(boolean schemaIdInHeader, KafkaCluster cluster, To
 
     @ParameterizedTest
     @ValueSource(booleans = { true, false })
-    void detectsClientProducingWithWrongSchemaId(boolean schemaIdInHeader, KafkaCluster cluster, Topic topic1) {
+    void clientSideUsesKeySchemasToo(boolean schemaIdInHeader, KafkaCluster cluster, Topic topic1) throws Exception {
+        var config = proxy(cluster)
+                .addToFilters(new FilterDefinitionBuilder(RecordValidation.class.getName()).withConfig("rules",
+                        List.of(Map.of("topicNames", List.of(topic1.name()), "keyRule",
+                                Map.of("schemaValidationConfig", Map.of("apicurioRegistryUrl", APICURIO_REGISTRY_URL, "apicurioGlobalId", firstGlobalId)))))
+                        .build());
+
+        var producerKeySerde = new JsonSchemaSerde<PersonBean>();
+        var valueSerde = new Serdes.StringSerde();
+
+        producerKeySerde.configure(Map.of(
+                SerdeConfig.REGISTRY_URL, APICURIO_REGISTRY_URL,
+                SerdeConfig.ARTIFACT_RESOLVER_STRATEGY, FixedArtifactReferenceResolver.class.getName(),
+                SerdeConfig.ENABLE_HEADERS, schemaIdInHeader), true);
+
+        var consumerKeySerde = new JsonSchemaSerde<>();
+        consumerKeySerde.configure(Map.of(
+                SerdeConfig.REGISTRY_URL, APICURIO_REGISTRY_URL), true);
```

Comment:
Could `producerKeySerde` and `consmerKeySerde` be pulled out as fields initialised in a `beforeEach`? 

Just feels like there is a lot of setup code obscuring what this test is trying to demonstrate.